firefox_puppeteer/api/software_update.py

```diff
@@ -0,0 +1,491 @@
+                                                                 attr))
+
+    @property
+    def exists(self):
+        """Is there an active update
+
+        :returns: True if there is an active update
+        """
+
+        active_update = self.marionette.execute_script("""
+          let ums = Cc['@mozilla.org/updates/update-manager;1']
+                    .getService(Ci.nsIUpdateManager);
+          return ums.activeUpdate;
+          """)
+
+        return bool(active_update)
+
+    def get_patch_at(self, patch_index):
+        """Use nsIUpdate.getPatchAt to return a patch from an update
+
+        :returns: JSON data for an nsIUpdatePatch object
+        """
+
+        return self.marionette.execute_script("""
+          let ums = Cc['@mozilla.org/updates/update-manager;1']
+                    .getService(Ci.nsIUpdateManager);
+          return ums.activeUpdate.getPatchAt(arguments[0]);
+          """, script_args=[patch_index])
+
+    @property
+    def patch_count(self):
+        """Get the patchCount from the active update
+
+        :returns: The patch count
+        """
+
+        return self.marionette.execute_script("""
+          let ums = Cc['@mozilla.org/updates/update-manager;1']
+                    .getService(Ci.nsIUpdateManager);
+          return ums.activeUpdate.patchCount;
+          """)
+
+    @property
+    def selected_patch(self):
+        """Get the selected patch for the active update
+
+        :returns: JSON data for the selected patch
+        """
+
+        return self.marionette.execute_script("""
+          let ums = Cc['@mozilla.org/updates/update-manager;1']
+                    .getService(Ci.nsIUpdateManager);
+          return ums.activeUpdate.selectedPatch;
+        """)
+
+
+class MARChannels(BaseLib):
+    """Class to handle the allowed MAR channels as listed in update-settings.ini
+    """
+
+    INI_SECTION = 'Settings'
+    INI_OPTION = 'ACCEPTED_MAR_CHANNEL_IDS'
+
+    def __init__(self, marionette_getter):
+        BaseLib.__init__(self, marionette_getter)
+
+        self.ini_file_path = self.marionette.execute_script("""
+          Cu.import('resource://gre/modules/Services.jsm');
+
+          let file = Services.dirsvc.get('GreD', Ci.nsIFile);
+          file.append('update-settings.ini');
+
+          return file.path;
+          """)
+
+    @property
+    def config_file_path(self):
+        return self.ini_file_path
+
+    @property
+    def config_file_contents(self):
+        with open(self.ini_file_path) as f:
+            return f.read()
+
+    @property
+    def channels(self):
+
+        channels = self.marionette.execute_script("""
+          Cu.import('resource://gre/modules/Services.jsm');
+
+          let iniFactory = Cc['@mozilla.org/xpcom/ini-processor-factory;1']
+                           .getService(Ci.nsIINIParserFactory);
+
+          let file = Services.dirsvc.get('GreD', Ci.nsIFile);
+          file.append('update-settings.ini');
```

Comment:
We have the path as `self.ini_file_path` so please push this in here as a script_arg.